rest/aggs/aggs_test.go

```diff
@@ -66,7 +66,39 @@ func TestGetAggs(t *testing.T) {
 	assert.Nil(t, err)
 	b, err := json.MarshalIndent(res, "", "\t")
 	assert.Nil(t, err)
-	assert.Equal(t, expectedResponse, string(b))
+	assert.Equal(t, expectedAggsResponse, string(b))
+}
+
+func TestGetAggsBuilder(t *testing.T) {
+	c := polygon.New("API_KEY")
+
+	httpmock.ActivateNonDefault(c.Aggs.HTTP.GetClient())
+	defer httpmock.DeactivateAndReset()
+	registerResponder("https://api.polygon.io/v2/aggs/ticker/AAPL/range/1/day/2021-07-22/2021-08-22?adjusted=true&customparam=customvalue&explain=false&limit=1&sort=desc", expectedAggsResponse)
+
+	res := &models.GetAggsResponse{}
+	req, err := c.NewRequest(context.Background(), models.GetAggsParams{
+		Ticker:     "AAPL",
+		Multiplier: 1,
+		Resolution: "day",
+		From:       time.Date(2021, 7, 22, 0, 0, 0, 0, time.UTC),
+		To:         time.Date(2021, 8, 22, 0, 0, 0, 0, time.UTC),
+		Adjusted:   models.Ptr(true),
+		Sort:       models.Ptr(models.Desc),
+		Limit:      models.Ptr(1),
+		Explain:    models.Ptr(false),
+	}, res)
+	// Allows using uncodumented params
+	req.SetQueryParam("customparam", "customvalue")
+	// ...or setting arbitrary headers
+	req.SetHeader("customheader", "customvalue")
+	assert.Nil(t, err)
+
+	_, err = req.Execute(http.MethodGet, models.GetAggsPath)
```

Comment:
it might be nice to build a request that already has these two inputs handled for the user so they could just call `req.Send()` once they're done adding custom stuff

Reply:
we discussed irl and this is probably not worth the extra work